### DISCUSSION TITLE
Installation on new Win10 Install fails with TLS/SSL Error

### DISCUSSION THREAD

**Discussion by rbharathkumar** [ScoopInstaller/Scoop#5319] [Help and Questions]
Hi All, hope you are all doing well.
I have used and enjoyed scoop very much and continue to do so.
But every time I install scoop on a new Win10 install, it fails with the TLS/SSL Error.

I see several similar issues dating back to the Win 8.1 days...
Have set the execution policy, latest PowerShell and PowerShell 5.1, latest and LTS dotnet runtimes, Git etc but to no avail....

I see the solution mentioned commonly as `[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12`

Just wanted to understand if other are facing the same issue...Cause of the issue, if it is widespread would it be possible to mention the `solution` in the advanced install instructions so that we can be sure about the command's safety instead of copy pasting some code from the comments.

Thanks for all the great work...
Cheers....

**Comment by rashil2000** [comment]
The installer handles this automatically, see https://github.com/ScoopInstaller/Install/blob/master/install.ps1#L157-L173

**Reply by rbharathkumar** [reply]
Thank you so much for your response....

Can you maybe guide me where I am going wrong? Please find the logs from both the default PowerShell and PowerShell 7.3.1.
MS .Net Core Runtime 3.1.32 and MS .Net Runtime 6.0.12 is installed. 

**PowerShell 7.3.1**
```pwsh
PS <Path>\Scoop> Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser
PS <Path>\Scoop> irm get.scoop.sh -outfile 'install.ps1'
Invoke-RestMethod: Authentication failed, see inner exception.
PS <Path>\Scoop> [System.Net.ServicePointManager]::SecurityProtocol
SystemDefault
```
**Default PowerShell**
```pwsh
PS <Path>\Scoop> Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser
PS <Path>\Scoop> irm get.scoop.sh -outfile 'install.ps1'
irm : The request was aborted: Could not create SSL/TLS secure channel.
At line:1 char:1
+ irm get.scoop.sh -outfile 'install.ps1'
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand
PS <Path>\Scoop> [System.Net.ServicePointManager]::SecurityProtocol
SystemDefault
```

**Reply by rashil2000** [reply]
What happens after you run `[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12` and try again?

**Reply by rbharathkumar** [reply]
have not run it yet - wanted to get the official opinion if it was safe to run that command...

**Reply by rashil2000** [reply]
It is safe to run.

**Reply by rbharathkumar** [reply]
Same Issue...

```pwsh
PowerShell 7.3.1
PS <Path>\Scoop> [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
PS <Path>\Scoop> [System.Net.ServicePointManager]::SecurityProtocol
Tls12
PS <Path>\Scoop> Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser
PS <Path>\Scoop> irm get.scoop.sh -outfile 'install.ps1'
Invoke-RestMethod: Authentication failed, see inner exception.
```

Can you also confirm if it will still allow my pc to use future versions of tls or other protocols and only limited to 1.2 ?

**Reply by rashil2000** [reply]
You can just revert back to the default using `[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::SystemDefault` if you face any problems.

Are you able to access `get.scoop.sh` in a browser?

**Reply by rbharathkumar** [reply]
In Edge - `get.scoop.sh` redirects to https://[raw.githubusercontent.com] and shows the below error message...
(https://raw.githubusercontent.com/scoopinstaller/install/master/install.ps1)/scoopinstaller/install/master/install.ps1

![image](https://user-images.githubusercontent.com/38585383/211133238-b858bfca-5e3d-4f78-b256-03fa643c35fa.png)

**Reply by rashil2000** [reply]
There seems to be some problem with your internet or location. Are you able to see this page: https://github.com/ScoopInstaller/Install/blob/master/install.ps1?

**Reply by chawyehsu** [reply]
Nothing needs to be done with `SecurityProtocol` when it is `SystemDefault`, it will use the proper protocol when it's available...

```powershell
function Optimize-SecurityProtocol {
    # .NET Framework 4.7+ has a default security protocol called 'SystemDefault',
    # which allows the operating system to choose the best protocol to use.
    # If SecurityProtocolType contains 'SystemDefault' (means .NET4.7+ detected)
    # and the value of SecurityProtocol is 'SystemDefault', just do nothing on SecurityProtocol,
    # 'SystemDefault' will use TLS 1.2 if the webrequest requires.
```

**Reply by rbharathkumar** [reply]
@rashil2000 yes, I am able to reach the link you had shared - 588 line install.ps1 script...
I also thought it was an internet issue, but I am able to reach the link though...
@chawyehsu thank you for your response - I have reverted back to SystemDefault using the command rashil2000 had provided....

**Reply by rashil2000** [reply]
> @rashil2000 yes, I am able to reach the link you had shared - 588 line install.ps1 script...

You can download that script file and run it in PowerShell, it will run the installer.

**Reply by rbharathkumar** [reply]
ok I had to copy the install.ps1 file manually and then I was able to install...
Weirdly, when I clicked on raw to copy the content, I once again received the same connection error...
What do you think is causing the error? the same as the screenshot above....

On a positive note, scoop is installed and working great...I appreciate and thank you for your all your guidance....

**Reply by rashil2000** [reply]
> What do you think is causing the error? the same as the screenshot above....

Frankly, I have no idea

**Reply by rbharathkumar** [reply]
that is ok - maybe if possible the instruction can be included in the advanced install instructions.
thank you once again for all the help...
